docs/basic/getting-started/react-prop-type-examples.md

```diff
@@ -13,7 +13,7 @@ export declare interface AppProps {
   functionChildren: (name: string) => React.ReactNode; // recommended function as a child render prop type
   style?: React.CSSProperties; // to pass through style props
   onChange?: React.FormEventHandler<HTMLInputElement>; // form events! the generic parameter is the type of event.target
-  props: Props & React.PropsWithoutRef<JSX.IntrinsicElements["button"]>; // to impersonate all the props of a button element without its ref
+  props: Props & React.ComponentPropsWithoutRef<"button">; // to impersonate all the props of a button element without its ref
```

Comment:
why not `ComponentProps` instead of `ComponentPropsWithoutRef`?

Reply:
oh... i just saw this https://github.com/DefinitelyTyped/DefinitelyTyped/blob/a05cc538a42243c632f054e42eab483ebf1560ab/types/react/index.d.ts#L761

Reply:
```suggestion
  props: Props & React.ComponentPropsWithoutRef<"button">; // to impersonate all the props of a button element without forwarding its ref. more info: https://react-typescript-cheatsheet.netlify.app/docs/advanced/patterns_by_usecase/#wrappingmirroring
```

Reply:
Maybe just being explicit:

```suggestion
  //  more info: https://react-typescript-cheatsheet.netlify.app/docs/advanced/patterns_by_usecase/#wrappingmirroring
  props: Props & React.ComponentPropsWithoutRef<"button">; // to impersonate all the props of a button element and explicitly not forwarding its ref
  props2: Props & React.ComponentPropsWithRef<MyButtonForwardedRef>; // to impersonate all the props of MyButtonForwardedRef and explicitly forwarding its ref
```